linode/networkingipassignment/framework_resource_schema.go

```diff
@@ -0,0 +1,38 @@
+package networkingipassignment
+
+import (
+	"github.com/hashicorp/terraform-plugin-framework/attr"
+	"github.com/hashicorp/terraform-plugin-framework/resource/schema"
+	"github.com/hashicorp/terraform-plugin-framework/resource/schema/listplanmodifier"
+	"github.com/hashicorp/terraform-plugin-framework/resource/schema/planmodifier"
+	"github.com/hashicorp/terraform-plugin-framework/resource/schema/stringplanmodifier"
+	"github.com/hashicorp/terraform-plugin-framework/types"
+)
+
+var frameworkResourceSchema = schema.Schema{
+	Attributes: map[string]schema.Attribute{
+		"id": schema.StringAttribute{
+			Computed:    true,
+			Description: "The ID of the IP assignment operation.",
+			PlanModifiers: []planmodifier.String{
+				stringplanmodifier.UseStateForUnknown(), // Use the state when ID is unknown.
+			},
+		},
+		"region": schema.StringAttribute{
+			Required:    true,
+			Description: "The region for the IP assignments.",
+		},
+		"assignments": schema.ListAttribute{
```

Comment:
I think it might make more sense to make one instance of this resource represent one assignment, similar to the [linode_placement_group_assignment](https://github.com/linode/terraform-provider-linode/blob/dev/linode/placementgroupassignment/framework_resource_schema.go) resource. In that case we would be able to use RequiresReplace, which would eliminate the need for update logic and would prevent the resource's ID from changing during updates, which seems to be disallowed by TF in my testing.

Reply:
I think having this resource support multiple assignments makes sense because the linode API for it works that way. I was wondering if we could just disallow a user from making any updates to this resource by throwing an error.